torch/fx/operator_schemas.py

```diff
@@ -0,0 +1,85 @@
+import torch
+import inspect
+import numbers
+import typing
+import enum
+from typing import Any, Callable, Dict, List, Optional
+
+_manual_overrides : Dict[Callable, List[inspect.Signature]] = {}
+
+def _nonzero_schemas():
+    signatures = []
+
+    def nonzero(self):
+        pass
+    signatures.append(inspect.signature(nonzero))
+
+    def nonzero(self, *, as_tuple : bool):  # type: ignore
+        pass
+    signatures.append(inspect.signature(nonzero))
+
+    return signatures
+
+_manual_overrides[torch.nonzero] = _nonzero_schemas()
+
+class _FakeGlobalNamespace:
+    def __getattr__(self, name):
+        if name == 'torch':
+            return torch
+        raise RuntimeError('Expected a torch namespace lookup lookup')
+
+def _torchscript_type_to_python_type(ts_type : 'torch._C.JitType') -> Any:
+    g = {'Tensor' : torch.Tensor, 'Device' : torch.device, 'Layout' : torch.layout,
+         'number' : numbers.Number, 'Future' : torch.jit.Future,
+         'AnyEnumType' : enum.Enum, 'QScheme' : torch.qscheme,
+         '__torch__': _FakeGlobalNamespace(),
+         't': typing.TypeVar('t')}  # type: ignore
+    for k in dir(typing):
```

Comment:
This loop seems a bit wasteful, it runs for every argument of function (and I assume for many functions). Then this list `g` will be constructed many many times. Since `g` doesn't change, can it be memoized?

Reply:
Sure, I think I can hoist this out as a private global. Just need to make sure `eval` doesn't mutate it